data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PluginSetting.java

```diff
@@ -136,6 +137,88 @@ public String getStringOrDefault(final String attribute, final String defaultVal
         throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
     }
 
+    /**
+     * Returns the value of the specified List<String>, or {@code defaultValue} if this settings contains no value for
+     * the attribute.
+     *
+     * @param attribute    name of the attribute
+     * @param defaultValue default value for the setting
+     * @return the value of the specified attribute, or {@code defaultValue} if this settings contains no value for
+     * the attribute
+     */
+    public List<String> getStringListOrDefault(final String attribute, final List<String> defaultValue) {
```

Comment:
This method and the other methods can probably be made somewhat more generic.

```
public <T> List<T> getTypedListOrDefault(final String attribute, final Class<T> type, final List<T> defaultValue)
```

You can replace your `instanceof` with `isAssignableFrom`.

Reply:
Had thought about this and wasn't sure if passing the Class<T> type was considered good practice. Doing this definitely scales PluginSettings better though.